### ISSUE TITLE
Merge docs for CheckInUtils withCheckIn and environment

### ISSUE THREAD

**Issue by adinauer** [getsentry/sentry-java#3902]
### Description

Merge this PR once v8 is ready / merge it into a v8 branch.

https://github.com/getsentry/sentry-docs/pull/11826#pullrequestreview-2444311698

**Comment by adinauer** [comment]
Merged into v8 collection branch, which is used in https://github.com/getsentry/sentry-docs/pull/11882
